### PR TITLE
add initial install artifacts w.r.t maya components

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This PR is the initial set of changes to give a home to artifacts related to maya resources. This also sets the context to make changes to cas template related artifacts for any corresponding change to cas engine in a single commit.

In near future:
- maya api server will be able to deploy these artifacts on a Kubernetes cluster in an automated manner.

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: install/openebs-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,2342 @@
...
+metadata:
+  name: cstorvolumes.openebs.io
+spec:
+  group: openebs.io
+  names:
+    kind: CStorVolume
+    listKind: CStorVolumeList
+    plural: cstorvolumes
+    shortNames:
+    - cstorvolume
```
Comment (princerachit) [inline install/openebs-operator.yaml]: maybe csv?

File: install/openebs-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,2342 @@
...
+      We pluck the cStorPool name from the map[uid]name:
+      { "uid1":"name1","uid2":"name2","uid2":"name2" }
+      The .ListItems.currentRepeatResource gives us the uid of one
+      of the pools from resources list
+      */}}
+      name: {{ .Volume.owner }}-{{ pluck .ListItems.currentRepeatResource .ListItems.cvolPoolList.pools | first }}
+      labels:
+        cstorpool.openebs.io/name: {{ pluck .ListItems.currentRepeatResource .ListItems.cvolPoolList.pools | first }}
+        cstorpool.openebs.io/uid: {{ .ListItems.currentRepeatResource }}
+        cstorvolume.openebs.io/uid: {{ .Volume.owner }}
```
Comment (princerachit) [inline install/openebs-operator.yaml]: We should update this and ask @gkGaneshR to make corresponsing changes in his code. This label name is incorrect.

File: install/openebs-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,2342 @@
...
+    apiVersion: openebs.io/v1alpha1
+    kind: CStorVolume
+    metadata:
+      name: {{ .Volume.owner }}
+      labels:
+        openebs.io/pv: {{ .Volume.owner }}
+    spec:
+      targetIP: {{ .TaskResult.cvolcreateputsvc.clusterIP }}
+      capacity: {{ .Volume.capacity }}
+      iqn: iqn.2016-09.com.openebs.cstor
```
Comment (princerachit) [inline install/openebs-operator.yaml]: The iqn should be uniform across all the yamls. I think the code should have the logic to split the iqn using ':' and utilize the first part. @Moteesh

File: install/openebs-operator.yaml (hunk 2 above)
Comment (gkGaneshR) [reply install/openebs-operator.yaml]: Do you mean to replace cstorvolumeCR.uid with cstorvolumeCR.Name ? If so, It is just a string change and no code change is required from my end.

Comment (princerachit) [reply install/openebs-operator.yaml]: Yes.

Comment (gkGaneshR) [reply install/openebs-operator.yaml]: it is only a string change for me, both will work fine in cstor-pool-mgmt container. No change is required from my end. Please discuss with @avishnu to confirm which one to use, coz. this needs to be common between control plane and data plane.

Comment (princerachit) [reply install/openebs-operator.yaml]: `cstorvolume.openebs.io/uid` -> `cstorvolume.openebs.io/name`

**Comment by kmova** [comment]
LGTM

**Pull request closed without merging**